docs/3.tutorials/auction/7-factory.md

```diff
@@ -0,0 +1,106 @@
+
+Luckily for us, there is already a [factory contract example](https://github.com/near-examples/factory-rust)! We will fork this example and slightly modify it to suit our use case. If you would like to learn more about how the factory contract works, you can take a look at the [associated documentation](https://docs.near.org/tutorials/examples/factory#generic-factory).
+
+The factory example only comes in rust since, currently, the JavaScript SDK does not allow you to embed the WASM file in the contract. This is a limitation of the SDK and not the blockchain itself.
+
+---
+
+## Changing the default contract
+
+In the current example, the factory contract deploys the donation contract example. We will change this to deploy our auction contract instead. 
+
+Firstly, we'll need the compiled auction contract WASM file. You can get this by running the following command in part four of `contract-rs`
+
+```
+cargo near build 
+```
+
+You will find the resulting WASM file in `target/near`; copy this file and use it to replace the WASM of the donation contract in the factory contract's source folder. Now edit the auction contract changing the path to the auction contract.
+
+<Language value="rust" language="rust">
+    <Github fname="Default init" 
+            url="https://github.com/near-examples/auctions-tutorial/blob/add-factory/factory/src/lib.rs#L9"
+            start="25" end="31" />
+    <Github fname="Contract path" 
+            url="https://github.com/near-examples/auctions-tutorial/blob/add-factory/factory/src/lib.rs#L9"
+            start="9" end="9" />
+</Language>
+
+On initialization, the factory will add the auction contracts WASM, as bytes, to the factory's state. It is more efficient to not store the WASM in the factory's state, however, we may want to update the auction contract if we find a bug or want to add new features. The factory implements a method to update the auction contract - we'll change the name to `update_auction_contract` as this factory will only deploy auction contracts.
+
+<Github fname="Contract path" language="rust"
+        url="https://github.com/near-examples/auctions-tutorial/blob/add-factory/factory/src/manager.rs#L8-L13"
+        start="8" end="13" />
+
+---
+
+## Modifying deploy method
+
+The method to deploy a new contract is specific to the contract being deployed (in the case the contract has custom initialization parameters). We will modify the method to take in the auction contract's initialization parameters.
+
+<Github fname="Contract path" language="rust"
+        url="https://github.com/near-examples/auctions-tutorial/blob/add-factory/factory/src/deploy.rs#L9-L82"
+        start="9" end="82" />
+
+In this fork, we have also removed the option to add an access key to the contract account since, as discussed in part 2, we want auctions to be locked.
```

Comment:
```suggestion
In this fork, we have also removed the option to add an access key to the contract account since, as discussed in [part 2](2-locking.md), we want auctions to be locked.
```